plotly/dashboard_objs/dashboard_objs.py

```diff
@@ -0,0 +1,302 @@
+
+username = plotly.tools.get_credentials_file()['username']
+api_key = plotly.tools.get_credentials_file()['api_key']
+headers = {'Plotly-Client-Platform': 'nteract'}
+
+
+class EmptyBox(dict):
+    def __init__(self):
+        self['type'] = 'box'
+        self['boxType'] = 'empty'
+
+
+class Box(dict):
+    def __init__(self, fileId='', shareKey=None, title=''):
+        self['type'] = 'box'
+        self['boxType'] = 'plot'
+        self['fileId'] = fileId
+        self['shareKey'] = shareKey
+        self['title'] = title
+
+
+class Container(dict):
+    def __init__(self, box_1=EmptyBox(), box_2=EmptyBox(), size=400,
+                 sizeUnit='px', direction='vertical'):
+        self['type'] = 'split'
+        self['size'] = size
+        self['sizeUnit'] = sizeUnit
+        self['direction'] = direction
+        self['first'] = box_1
+        self['second'] = box_2
+
+
+class Dashboard(dict):
+    def __init__(self, dashboard_json=None, backgroundColor='#FFFFFF',
+                 boxBackgroundColor='#ffffff', boxBorderColor='#d8d8d8',
+                 boxHeaderBackgroundColor='#f8f8f8', foregroundColor='#333333',
+                 headerBackgroundColor='#2E3A46', headerForegroundColor='#FFFFFF',
+                 links=[], logoUrl='', title='Untitled Dashboard'):
+        self.box_ids_dict = {}
+        if not dashboard_json:
+            self['layout'] = EmptyBox()
+            self['version'] = 2
+            self['settings'] = {
+                'backgroundColor': backgroundColor,
+                'boxBackgroundColor': boxBackgroundColor,
+                'boxBorderColor': boxBorderColor,
+                'boxHeaderBackgroundColor': boxHeaderBackgroundColor,
+                'foregroundColor': foregroundColor,
+                'headerBackgroundColor': headerBackgroundColor,
+                'headerForegroundColor': headerForegroundColor,
+                'links': links,
+                'logoUrl': logoUrl,
+                'title': title
+            }
+            # TODO: change name to box_id_to_path
+        else:
+            self['layout'] = dashboard_json['layout']
+            self['version'] = dashboard_json['layout']
+            self['settings'] = dashboard_json['settings']
+
+            all_nodes = []
+            node_gen = node_generator(dashboard_json['layout'])
+
+            finished_iteration = False
+            while not finished_iteration:
+                try:
+                    all_nodes.append(node_gen.next())
+                except StopIteration:
+                    finished_iteration = True
+
+            for node in all_nodes:
+                if (node[1] != () and node[0]['type'] == 'box' and
+                    node[0]['boxType'] != 'empty'):
+                    try:
+                        max_id = max(self.box_ids_dict.keys())
+                    except ValueError:
+                        max_id = 0
+                    self.box_ids_dict[max_id + 1] = list(node[1])
```

Comment:
That's probably a better idea since that's exactly what I'm doing anyways when I pull a dashboard from online. I have code in here for dealing with that so reusing the same code would probably be best. So a big :+1: from me